apps/omg/test/omg/state/fee_test.exs

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.State.FeeTest do
+  @moduledoc false
+
+  use ExUnitFixtures
+  use ExUnit.Case, async: true
+
+  alias OMG.State.Transaction
+
+  @eth OMG.Eth.RootChain.eth_pseudo_address()
+
+  @tag fixtures: [:alice]
+  test "can be encoded to binary form and back", %{alice: owner} do
+    fee_tx = Transaction.Fee.new(1000, {owner.addr, @eth, 1551})
+
+    rlp_form = Transaction.raw_txbytes(fee_tx)
+    assert fee_tx == Transaction.decode!(rlp_form)
+  end
+
```

Comment:
where would adding tests like ones living in `OMG.State.TransactionTest` (those tests are for `Payment` only so the name is not perfect) belong?

I think now, since the encoding/decoding code exists, is the time to figure those tests out.

Reply:
Yeah, I've noticed that `OMG.State.TransactionTest` entirerly tests encoding / recovering scenarios. Mixing `FeeTokenClaim` with them isn't best idea (is it?)
Maybe It should go to `transaction/payment_test.exs` and these to `transaction/fee_token_claim_test.exs` :question:

Reply:
yeah, this sounds reasonable. Note that that big test has undergone a split already, between a more top-level `Recovered` tests and lower level `Transaction` tests. But I think the approach is still valid and feasible